actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserSelfDeclarationManagementActor.java

```diff
@@ -1,299 +1,140 @@
 package org.sunbird.user.actors;
 
-import com.fasterxml.jackson.core.type.TypeReference;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import java.sql.Timestamp;
 import java.util.*;
 import org.apache.commons.collections.CollectionUtils;
-import org.apache.commons.collections4.MapUtils;
 import org.apache.commons.lang3.StringUtils;
 import org.sunbird.actor.core.BaseActor;
 import org.sunbird.actor.router.ActorConfig;
-import org.sunbird.cassandra.CassandraOperation;
 import org.sunbird.common.exception.ProjectCommonException;
 import org.sunbird.common.models.response.Response;
 import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.TelemetryEnvKey;
 import org.sunbird.common.request.Request;
 import org.sunbird.common.request.RequestContext;
 import org.sunbird.common.responsecode.ResponseCode;
-import org.sunbird.helper.ServiceFactory;
+import org.sunbird.learner.util.Util;
 import org.sunbird.models.user.UserDeclareEntity;
-import org.sunbird.user.util.UserActorOperations;
+import org.sunbird.telemetry.util.TelemetryUtil;
+import org.sunbird.user.service.UserSelfDeclarationService;
+import org.sunbird.user.service.impl.UserSelfDeclarationServiceImpl;
+import org.sunbird.user.util.UserUtil;
 
 @ActorConfig(
   tasks = {"upsertUserSelfDeclarations"},
   asyncTasks = {"upsertUserSelfDeclarations", "updateUserSelfDeclarationsErrorType"}
 )
 public class UserSelfDeclarationManagementActor extends BaseActor {
-  private CassandraOperation cassandraOperation = ServiceFactory.getInstance();
-  private ObjectMapper mapper = new ObjectMapper();
+  private UserSelfDeclarationService userSelfDeclarationService =
+      UserSelfDeclarationServiceImpl.getInstance();
 
   @Override
   public void onReceive(Request request) throws Throwable {
-    if (UserActorOperations.UPSERT_USER_SELF_DECLARATIONS
-        .getValue()
-        .equalsIgnoreCase(request.getOperation())) {
-      upsertUserSelfDeclaredDetails(request);
-    }
-    if (UserActorOperations.UPDATE_USER_SELF_DECLARATIONS_ERROR_TYPE
-        .getValue()
-        .equalsIgnoreCase(request.getOperation())) {
-      updateUserSelfDeclaredErrorStatus(request);
-    } else {
-      onReceiveUnsupportedOperation("upsertUserSelfDeclarations");
+    String operation = request.getOperation();
+    switch (operation) {
+      case "updateUserDeclarations": // update self declare
```

Comment:
Retain with the constants in the switch cases

Reply:
Cannot retain constants as ActorOperations is an enum class. All the actors follow same logic as of now.